mantidproject/mslice
# Non-PSD Mode

This PR adds a different workflow for non-PSD type data (from MARI or from other instruments using a `rings_map`). These data are defined by having detector lists in order sorted by `2theta` angles. The main characteristics is that there are few detectors in these data, and that centre-point rebinning (using  `ConvertToMD`) is not suitable. Until now, the work-around has been to use `SofQW3` to rebin the data into fine `|Q|` bins and then to convert to MD. However, this two step binning process overestimates the uncertainty (errors), because the first step involves partitioning bins. It would thus be better if the workflow includes only a single rebin.

This PR does this. Non-PSD data now do not need to go through a `Calculate Projection` step (i.e. conversion to `MDEventWorkspace`). `SofQW3` is used directly to rebin the data into a 2D slice in `|Q|` and `DeltaE` *and* also 1D cuts in these dimensions. For slices in `Degrees` (or `theta`), `ConvertSpectrumAxis` and `Rebin2D` is used (similar to the old workflow where `ConvertSpectrumAxis`, `ConvertToMD` and `BinMD` was used). 

**To test:**

Load a non-PSD input data file (e.g. `MAR21335_Ei60.00meV.nxs` or `MER30936_Ei10.00meV_rings.nxs`). Select the loaded workspace and check that the `Powder` tab is *disabled* and `Slice` and `Cut` *enabled*. Make slices and cuts as normal - check all supported features (recoil lines, powder lines, change intensity, interactive cuts, simultaneous cuts of multiple workspaces, cuts with integration widths) work as expected. 

Load a PSD input data file (e.g. `MER30936_Ei10.00meV_one2one.nxs`). Select the loaded workspace and check that `Powder` is *enabled* and `Slice` and `Cut` is *disabled*. Calculate projections and make cuts / slices / etc. and check everything works as before.


Fixes #254 .
